internal/repository/rsl.go

```diff
@@ -90,6 +90,23 @@ func (r *Repository) RecordRSLAnnotation(rslEntryIDs []string, skip bool, messag
 	return rsl.NewAnnotationEntry(rslEntryHashes, skip, message).Commit(r.r, signCommit)
 }
 
+// InvalidateLatestRSLEntry is used when the latest RSL entry is invalid.
+// This does not skip the RSL entry with an annotation, instead it resets
+// the ref, rolling it back by one commit.
+func (r *Repository) InvalidateLatestRSLEntry() error {
+	latestEntry, err := rsl.GetLatestEntry(r.r)
+	if err != nil {
+		return err
+	}
+
+	v, err := rsl.GetParentForEntry(r.r, latestEntry)
+	if err != nil {
+		return err
+	}
+
+	return r.r.Storer.SetReference(plumbing.NewHashReference(rsl.Ref, v.GetID()))
```

Comment:
gitinterface.ResetCommit?

Reply:
/ ResetDueToError

Reply:
Hm, that may have issues because it checks out the specified ref. This might be easier with the git binary support :thinking:

Reply:
I think it should currently be left like this, since we should think about the real functionality of gitinterface.ResetCommit, should we be moving the HEAD pointer of the reference pointer?